### PR TITLE
Add more vrrp tests

### PR DESCRIPTION
**Pull request by SerhiyBoikoPLV**
- test_vrrp_and_stp
- test_vrrp_preempt_on[bridge]
- test_vrrp_preempt_on[port]
- test_vrrp_under_traffic[bridge]
- test_vrrp_under_traffic[port]
- test_vrrp_under_traffic[vlan]
- test_vrrp_multiple_addr
- test_vrrp_max_instances
- test_vrrp_master_and_backup

Update setup_topo_for_vrrp function, which will allow to use:
- TG - DUT links
- links between infra devices
- VRRP over bridge or port or bridge vlan interfaces

Update verify_vrrp_ping function:
- Count only icmp reply packets.
- Pass interval as an argument.
- Add an argument to run tcpdump without running ping.

Add a generic polling function.

Depends on https://github.com/PLVision/dent-testing/pull/134

Signed-off-by: Serhiy Boiko <serhiy.boiko@plvision.eu>

### CODE REVIEW

File: DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vrrp/test_vrrp_advert.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
+from operator import itemgetter
+from math import isclose as is_close
+import asyncio
+import pytest
+
+from dent_os_testbed.lib.tc.tc_filter import TcFilter
+from dent_os_testbed.lib.tc.tc_qdisc import TcQdisc
+
+from dent_os_testbed.test.test_suite.functional.vrrp.vrrp_utils import (
+    setup_topo_for_vrrp,
```
Comment (StepanVovkPLV) [inline DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vrrp/test_vrrp_advert.py:10]: do u need `()` for 1 import

File: DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vrrp/test_vrrp_advert.py [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+
+    # 3. Configure VRRP on infra devices
+    vrrp_ifaces = [bridge, bridge]
+    advert_int = 1
+    await asyncio.gather(*[
+        configure_vrrp(dent, state=state, prio=prio, vr_ip=vrrp_ip, vr_id=vr_id,
+                       dev=port, additional_opts=[f'advert_int {advert_int}'])
+        for dent, port, state, prio
+        in zip(infra, vrrp_ifaces, ['MASTER', 'BACKUP'], [200, 100])])
+    await asyncio.sleep(10)  # wait for keepalived to start
```
Comment (StepanVovkPLV) [inline DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vrrp/test_vrrp_advert.py]: use variable for sleep? 
wait for keepalived to start  on 81, 121 lines

**Review by AndriyLozovyyPLV — APPROVED** [review]

File: DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vrrp/test_vrrp_advert.py (hunk 2 above)
Comment (SerhiyBoikoPLV) [reply DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vrrp/test_vrrp_advert.py]: done

File: DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vrrp/test_vrrp_advert.py (hunk 1 above)
Comment (SerhiyBoikoPLV) [reply DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vrrp/test_vrrp_advert.py:10]: this is to be consistent with other tests and for any future update (which will look better in diff)

**Comment by SerhiyBoikoPLV** [comment]
closed in favor of https://github.com/PLVision/dent-testing/pull/166

**Pull request closed without merging**